From 4acf6ad2a745bb54b5749b71e5c5c23ceb0b3f20 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Sat, 24 Jan 2015 10:36:54 +0800
Subject: [PATCH] MLK-10244-8 ARM: dts: imx7d-12x12-arm: add ecspi3 support

Enable ecspi3 support

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit 08d21459bd7d2ab7d6cde7bb7fb1c3ba1d7b9365)
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index 287bdd0..3f275ab 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -41,6 +41,22 @@
 	};
 };
 
+&ecspi3 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <&gpio6 22 0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi3_1 &pinctrl_ecspi3_cs_1>;
+	status = "okay";
+
+	flash: m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "st,m25p32";
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+	};
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
@@ -152,6 +168,21 @@
 };
 
 &iomuxc {
+	ecspi3 {
+		pinctrl_ecspi3_cs_1: ecspi3_cs_grp-1 {
+			fsl,pins = <
+				MX7D_PAD_SAI2_TX_DATA__GPIO6_IO22	0x2
+			>;
+		};
+
+		pinctrl_ecspi3_1: ecspi3grp-1 {
+			fsl,pins = <
+				MX7D_PAD_SAI2_TX_SYNC__ECSPI3_MISO	0x2
+				MX7D_PAD_SAI2_TX_BCLK__ECSPI3_MOSI	0x2
+				MX7D_PAD_SAI2_RX_DATA__ECSPI3_SCLK	0x2
+			>;
+		};
+	};
 
 	enet1 {
 		pinctrl_enet1: enet1grp {
-- 
1.8.0

